### PR TITLE
Attach wkconnect

### LINKED ISSUE #3952: wkconnect is unreachable from wk in the docker-compose setup

### PR DESCRIPTION
**Pull request by normanrz**
- Exposes wkconnect in the same container as webknossos by using `network_mode: service:webknossos`
- Replaces `links` with `depends_on` because they are deprecated

### Steps to test:
- `./start-docker.sh`

### Issues:
- fixes #3952

------
~- [ ] Updated [changelog](../blob/master/CHANGELOG.md#unreleased)~
~- [ ] Updated [migration guide](../blob/master/MIGRATIONS.md#unreleased) if applicable~
~- [ ] Updated [documentation](../blob/master/docs) if applicable~
~- [ ] Adapted [wk-connect](https://github.com/scalableminds/webknossos-connect) if datastore API changes~
~- [ ] Needs datastore update after deployment~
- [x] Ready for review

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -209,7 +202,8 @@ services:
     image: scalableminds/webknossos-connect:master__190
     volumes:
       - "./conf/connect:/app/data"
-    network_mode: host
```
Comment (jstriebel) [inline docker-compose.yml]: This was necessary to make it work with `yarn start`. We could use an env var to overwrite it in the `proxy.js`.

Comment (normanrz) [reply docker-compose.yml]: Seems to me this integration of wkconnect is turning into a complexity nightmare 🤦‍♂️ 
I think `yarn start` should not depend on docker at all. Submodules are also annoying. Forcing people to have the right Python version and deps seems also a bit too much. Perhaps we remove wkconnect from `yarn start`?

Comment (jstriebel) [reply docker-compose.yml]: I think for this integration it is perfectly fine to use docker, we all have it anyways. Why should `yarn start` not rely on docker?
We can either have the network overwritten by an env var or just do `docker run --rm -v … --network host scalableminds/webknossos-connect:…`. I think it makes sense to couple both closer without forcing one to install all the python deps.

Comment (normanrz) [reply docker-compose.yml]: The problem is that Docker is incredibly resource-hungry on any non-Linux platform. When developing I wouldn't want to be forced to have it running. I see wkconnect more as an addon than a core component.

Comment (jstriebel) [reply docker-compose.yml]: Ok, I just removed wkconnect from the proxy.js.

**Comment by jstriebel** [comment]
Also now the default `application.insertLocalConnectDatastore` config should probably be false, and only true for the `webknossos` service.

**Comment by normanrz** [comment]
I can't reproduce the `forbidden` on my machine.

**Pull request merged**